pkg/driver/aci_driver.go

```diff
@@ -0,0 +1,760 @@
+		if err != nil {
+			return fmt.Errorf("Attempt to List Subscriptions Failed: %v", err)
+		}
+
+		err = result.NextWithContext(ctx)
+		if err != nil {
+			return fmt.Errorf("Attempt to Get Subscription Failed: %v", err)
+		}
+
+		// Just choose the first subscription
+		result.NextWithContext(ctx)
+		if result.NotDone() {
+			subscriptionID := *result.Value().SubscriptionID
+			d.log("Setting Subscription to", subscriptionID)
+			d.subscriptionID = subscriptionID
+		} else {
+			return errors.New("Cannot find a subscription")
+		}
+
+	}
+	return nil
+}
+
+func (d *ACIDriver) createACIInstance(op *Operation) error {
+	// GET ACI Config
+	aciRG := d.config["ACI_RESOURCE_GROUP"]
+	aciLocation := d.config["ACI_LOCATION"]
+	aciLocation = strings.ToLower(strings.Replace(aciLocation, " ", "", -1))
+	d.log("Resource Group:", aciRG)
+	d.log("Location:", aciLocation)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	if len(aciLocation) > 0 {
+		providersClient := d.getProvidersClient()
+		provider, err := providersClient.Get(ctx, "Microsoft.ContainerInstance", "")
+		if err != nil {
+			return fmt.Errorf("Error getting provider details for ACI: %v", err)
+		}
+
+		for _, t := range *provider.ResourceTypes {
+			if *t.ResourceType == "ContainerGroups" {
+				if !locationIsAvailable(aciLocation, *t.Locations) {
+					return fmt.Errorf("ACI Location is invalid: %v", aciLocation)
+				}
+			}
+		}
+
+	} else if len(aciRG) == 0 {
+		return errors.New("ACI Driver requires ACI_LOCATION environment variable or an existing Resource Group in ACI_RESOURCE_GROUP")
+	}
+
+	groupsClient := d.getGroupsClient()
+	if len(aciRG) == 0 {
+		d.log("Creating Resource Group")
+		aciRG = uuid.New().String()
+		d.log("Resource Group:", aciRG)
+		_, err := groupsClient.CreateOrUpdate(
+			ctx,
+			aciRG,
+			resources.Group{
+				Location: &aciLocation,
+			})
+		if err != nil {
+			return fmt.Errorf("Failed to create resource group: %v", err)
+		}
+
+		defer func() {
+			if d.deleteACIResources {
+				d.log("Deleting Resource Group")
+				future, err := groupsClient.Delete(ctx, aciRG)
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "Failed to execute delete resource group %s error: %v\n", aciRG, err)
+				}
+
+				err = future.WaitForCompletionRef(ctx, groupsClient.Client)
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "Failed to delete resource group %s error: %v\n", aciRG, err)
+				} else {
+					d.log("Deleted Resource Group ", aciRG)
+				}
+
+			}
+
+		}()
+	} else {
+		_, err := groupsClient.Get(ctx, aciRG)
+		if err != nil {
+			return fmt.Errorf("Checking for existing resource group %s failed with error: %v", aciRG, err)
+		}
+
+	}
+
+	aciName := d.config["ACI_NAME"]
+	d.log("ACI Name:", aciName)
+	if len(aciName) == 0 {
+		aciName = fmt.Sprintf("duffle-%s", uuid.New().String())
+		d.log("ACI Name:", aciName)
+	}
+
+	// TODO ACI does not support file copy
+	// Does not support files because ACI does not support file copy yet
```

Comment:
Would this be avoided by creating a file share and mounting it?

(Not saying it should be part of this PR, just for future reference.)

Reply:
I don't think that this can be resolved using file mount because the data the needs to be written to the file may be sensitive, also its not possible to mount a file into a pre-existing folder without masking the contents of the folder in the image